flink-tests/src/test/java/org/apache/flink/test/checkpointing/ResumeCheckpointManuallyITCase.java

```diff
@@ -263,56 +268,113 @@ private void testExternalizedCheckpoints(
 			config.setString(HighAvailabilityOptions.HA_STORAGE_PATH, haDir.toURI().toString());
 		}
 
-		TestingCluster cluster = new TestingCluster(config);
-		cluster.start();
+		MiniClusterResource cluster = new MiniClusterResource(
+			new MiniClusterResource.MiniClusterResourceConfiguration(
+				new Configuration(),
+				NUM_TASK_MANAGERS,
+				SLOTS_PER_TASK_MANAGER),
+			true);
```

Comment:
Can we make this a `@ClassRule`?

Reply:
ha. It _looks_ like it could be, but only because I did a mistake. We're passing a blank configuration to the cluster, and not the `config` we set up above.

With that fixed the can't be a class rule since each test requires it's own configuration. It also cannot be turned into a `Rule` easily as the config parameters are passed through the methods, and we would have to restructure the tests rather significantly.